webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAIPromptService.java

```diff
@@ -0,0 +1,169 @@
+            .orElseThrow(() -> new AIException("Prompt not found: " + promptId));
+    aiPrompt.setDeleted(true);
+    aiPrompt.setLastModifiedDate(JSR310Migration.dateTimeNow());
+    aiPromptRepository.save(aiPrompt);
+  }
+
+  @Timed("OpenAIPromptService.getPrompt")
+  public AIPrompt getPrompt(Long promptId) {
+    return aiPromptRepository
+        .findById(promptId)
+        .orElseThrow(() -> new AIException("Prompt not found: " + promptId));
+  }
+
+  @Timed("OpenAIPromptService.getAllActivePrompts")
+  public List<AIPrompt> getAllActivePrompts() {
+    return aiPromptRepository.findByDeletedFalse();
+  }
+
+  @Override
+  @Transactional
+  @Timed("OpenAIPromptService.createPromptContextMessage")
+  public Long createPromptContextMessage(
+      AIPromptContextMessageCreateRequest aiPromptContextMessageCreateRequest) {
+    AIPromptContextMessage aiPromptContextMessage = new AIPromptContextMessage();
+    OpenAIPromptContextMessageType messageType =
+        OpenAIPromptContextMessageType.valueOf(
+            aiPromptContextMessageCreateRequest.getMessageType().toUpperCase());
+    aiPromptContextMessage.setContent(aiPromptContextMessageCreateRequest.getContent());
+    aiPromptContextMessage.setAiPrompt(
+        aiPromptRepository
+            .findById(aiPromptContextMessageCreateRequest.getAiPromptId())
+            .orElseThrow(
+                () ->
+                    new AIException(
+                        "Prompt not found: "
+                            + aiPromptContextMessageCreateRequest.getAiPromptId())));
+    aiPromptContextMessageRepository
+        .findByAiPromptIdAndOrderIndexAndDeleted(
+            aiPromptContextMessageCreateRequest.getAiPromptId(),
+            aiPromptContextMessageCreateRequest.getOrderIndex(),
+            false)
+        .ifPresent(
+            existingMessage -> {
+              throw new AIException(
+                  "Prompt context message already exists for order index: "
+                      + aiPromptContextMessageCreateRequest.getOrderIndex());
+            });
+    aiPromptContextMessage.setOrderIndex(aiPromptContextMessageCreateRequest.getOrderIndex());
+    aiPromptContextMessage.setMessageType(messageType.getType());
+    ZonedDateTime now = JSR310Migration.dateTimeNow();
+    aiPromptContextMessage.setCreatedDate(now);
+    aiPromptContextMessage.setLastModifiedDate(now);
+    return aiPromptContextMessageRepository.save(aiPromptContextMessage).getId();
+  }
+
+  @Timed("OpenAIPromptService.deletePromptContextMessage")
+  public void deletePromptContextMessage(Long promptMessageId) {
+    AIPromptContextMessage aiPromptContextMessage =
+        aiPromptContextMessageRepository
+            .findById(promptMessageId)
+            .orElseThrow(
+                () -> new AIException("Prompt context message not found: " + promptMessageId));
+    aiPromptContextMessage.setDeleted(true);
+    aiPromptContextMessage.setLastModifiedDate(JSR310Migration.dateTimeNow());
```

Comment:
Technically not needed - as the table as you've defined it would provide a value for the field.